api/v1/putRoutes.php

```diff
@@ -373,6 +373,67 @@
 	echoResponse( $r );
 });
 
+
+
+//
+//	URL:	/api/v1/deviceport/
+//	Method:	PUT
+//	Params: 
+//	Returns: record as created
+//
+
+$app->put( '/deviceport', function() {
+
+// DevicePort->updatePort() takes care of all this:
+// - verify both devices exist
+// - verify rights to both devices
+// -verify both ports exist
+// - update database, including deleting both old connections
+
+/*
+    Example payload:
+      "DeviceID": "1", //mandatory
+      "PortNumber": "1", //mandatory
+      "Label": "Port1",
+      "MediaID": "0",
+      "ColorID": "0",
+      "ConnectedDeviceID": null,
+      "ConnectedPort": null,
+      "Notes": ""
+*/
+// so all we need to do is populate a new DevicePorts object for the 
+// A-side port, populate its attributes
+// and call updatePort(); if it returns false then there was an error
+// It is expected that this would overwrite the entire existing port
+// due to being a PUT instead of POST
+	$dp = new DevicePorts();
+	$vars = getParsedBody();
+	foreach($vars as $prop => $val){
+		if ( property_exists($dp, $prop)) {
+			$dp->$prop=$val;
+		}
+	}
+	$status = $dp->updatePort();
+	if (! $status) {
+		$r['error'] = true;
+		$r['errorcode'] = 400;
+		$r['message'] = __("Port update failed");
```

Comment:
I did not immediately have a good idea for how to make this error message more useful, since dp->updatePort() just returns a boolean status